src/Sulu/Bundle/RouteBundle/Generator/TranslatorWrapper.php

```diff
@@ -41,6 +38,11 @@ public function setLocale(string $locale): void
 
     public function getLocale(): string
     {
+        // This case should not exist, only with the pseudo locale provider, and it doesn't have a locale.
+        if (!\method_exists($this->translator, 'getLocale')) {
+            return 'en';
+        }
```

Comment:
Don't get this. translatorinterface requires getLocale to exist how is this happening?

Reply:
is there any real case where you find that the getLocale failed?

Reply:
This is only a fix for Sulu 2.x. In Sulu 3 we are requiring Symfony version 6+ and there the interface has the method and we can remove it. Furthermore we can even use the LocalizationSwitcher. 

The fix there only affects the PseudolocationTranslator (which doesn't need a locale) so hardcoding anything is fine in my opinion.

Reply:
```
sulu/article-bundle                       2.6.0             Bundle for managing localized content-rich entities like blog-posts in the Sulu content management system
sulu/community-bundle                     3.0.x-dev 7a4e537 Sulu Bundle for building a community
sulu/sulu                                 2.5.19            Core framework that implements the functionality of the Sulu content management system
symfony/framework-bundle                  6.3.12            Provides a tight integration between Symfony components and the Symfony full-stack framework
```
But I just check the stack traces they're both in the sulu/sulu one in the website package and the other one in the routing bundle.

Reply:
which class you get into the service? Which version of the symfony translation contracts is installed.

The 6.3.12 even has the getLocale inside the interface: https://github.com/symfony/symfony/blob/v6.3.12/src/Symfony/Contracts/Translation/TranslatorInterface.php

Reply:
Ah yes, you are correct. It's getting the Symfony\Component\Translation\PseudoLocalizationTranslator and while it does have a 'getLocale' function (at least in 6.3+). It doesn't have a setLocale. That's what's causing the crash.